vpdq/cpp/regtest.py

```diff
@@ -72,91 +74,114 @@ def get_argparse() -> argparse.ArgumentParser:
     downsampleFrameDimension = str(args.downsampleFrameDimension)
-    verbose = args.verbose
-    # TODO: Add more general options for other video encodings.
-    for file in os.listdir(inputVideoFolder):
-        if file.endswith(".mp4"):
-            if verbose:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-v",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-
-    cdir = os.getcwd()
-    pdir = os.path.dirname(cdir)
-    sample = pdir + ("/sample-hashes")
-    output = outputHashFolder
     distanceTolerance = str(args.matchDistanceTolerance)
     qualityTolerance = str(args.qualityTolerance)
-    for file in os.listdir(sample):
-        if file.endswith(".txt"):
-            print("\nMatching File " + file)
-            sampleFile = f"{sample}/{file}"
-            outputFile = f"{output}/{file}"
+    verbose = args.verbose
+
+    if not inputVideoFolder.exists():
+        raise argparse.ArgumentTypeError(
+            f"inputVideoFolder: {inputVideoFolder} does not exist."
+        )
+
+    with TemporaryDirectory() as tempOutputHashFolder:
+        """
+        Write the files to temp
+        If outputHashFolder is specified then
+        copy the output files there, overwriting existing files
+        """
+
+        tempOutputHashFolder = Path(tempOutputHashFolder)
+        if outputHashFolder is not None:
+            if not outputHashFolder.exists():
+                print(f"Creating output directory at {outputHashFolder}")
+                outputHashFolder.mkdir(parents=True, exist_ok=True)
+            print(f"Writing output to directory: {outputHashFolder}")
+        else:
+            print(f"Writing output to temp directory: {tempOutputHashFolder}")
+        # TODO: Add more general options for other video extensions.
+        for fileStr in glob.iglob(f"{inputVideoFolder}/**/*.mp4", recursive=True):
+            file = Path(fileStr)
+
+            """
+            Create output hash file or overwrite existing file
```

Comment:
No. copyfile can overwrite existing files with the same name if you specify the output directly. Otherwise you have to delete the files every run if you use -d which is a huge pain.